src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs

```diff
@@ -1224,14 +1239,34 @@ internal ReadOnlySpan<byte> ValueSpan
             }
         }
 
-        // TODO make public https://github.com/dotnet/runtime/issues/33388
-        internal static bool DeepEquals(JsonElement left, JsonElement right)
+        /// <summary>
+        /// Compares the values of two <see cref="JsonElement"/> values for equality, including the values of all descendant elements.
+        /// </summary>
+        /// <param name="element1">The <see cref="JsonElement"/> to compare.</param>
+        /// <param name="element2">The <see cref="JsonElement"/> to compare.</param>
+        /// <returns><see langword="true"/> if the two values are equal; otherwise <see langword="false"/>.</returns>
+        /// <remarks>
+        /// Deep equality of two JSON values is defined as follows:
+        /// <list type="bullet">
+        /// <item>JSON values of different kind are not equal.</item>
+        /// <item>JSON constants <see langword="null"/>, <see langword="false"/> and <see langword="true"/> only equal themselves.</item>
+        /// <item>JSON numbers are equal if and only if they have they have equivalent decimal representations, with no rounding being used.</item>
+        /// <item>JSON strings are equal if and only if they are equal using ordinal string comparison.</item>
+        /// <item>JSON arrays are equal if and only if they are of equal length and each of their elements are pairwise equal.</item>
+        /// <item>
+        ///     JSON objects are equal if and only if they have the same number of properties and each property in the first object
+        ///     has a corresponding property in the second object with the same name and equal value. The order of properties is not
+        ///     significant, with the exception of repeated properties that must be specified in the same order (with interleaving allowed).
+        /// </item>
+        /// </list>
```

Comment:
Any exceptions to document, e.g. from the CheckValidInstance calls?

Reply:
It's not being documented in other `JsonElement` APIs either https://github.com/dotnet/runtime/pull/104579#discussion_r1675942717.